File: backend/src/main/java/com/trackflow/backend/entities/Employee.java
package com.trackflow.backend.entities;

//import java.util.Date;
//import jakarta.persistence.*;
//import lombok.*;
//
//@Setter
//@Getter
//@NoArgsConstructor
//@AllArgsConstructor
//@Entity
//public class Employee {
//	@Id
//	@GeneratedValue(strategy = GenerationType.IDENTITY)
//	int empId;
//	
//	@Column
//	String fullName;
//	
//	@Column
//	Date dob;
//	
//	@Column
//	String gender;
//	
//	@Column
//	String nationality;
//
//	@Column
//	String email;
//		
//	@Column
//	String phNo;
//	
//	@Column
//	String currentAddress;
//	
//	@Column
//	String permanentAddress;
//	
//	@Column
//	double basicSal;
//	
//	@Column
//	Date hireDate;
//	
//	@Column
//	double incentives;
//	
//	@OneToOne
//	@JoinColumn(name = "userId")
//	User user;
//	
//	@ManyToOne
//	@JoinColumn(name = "designationID")
//	Designation designation;
//	
//	@Column
//	Role roleId;
//	
//	@Column
//	Designation designationID;
//	
//}

File: DotNetApi/TrackFlowDotNetRestApi/Models/Employee.cs
﻿using System;
using System.Collections.Generic;

namespace TrackFlowDotNetRestApi.Models
{
    public partial class Employee
    {
        public Employee()
        {
            Projects = new HashSet<Project>();
            Tasks = new HashSet<Task>();
        }

        public int Empid { get; set; }
        public string? Fullname { get; set; }
        public DateTime? Dob { get; set; }
        public string? Gender { get; set; }
        public string? Nationality { get; set; }
        public string? Email { get; set; }
        public string? Contact { get; set; }
        public string? Currentaddress { get; set; }
        public string? Permanentaddress { get; set; }
        public double? Basicsal { get; set; }
        public float? Incentives { get; set; }
        public DateTime? Hiredate { get; set; }
        public int? DesignationId { get; set; }
        public int? LoginId { get; set; }

        public virtual Designation? Designation { get; set; }
        public virtual Login? Login { get; set; }
        public virtual ICollection<Project> Projects { get; set; }
        public virtual ICollection<Task> Tasks { get; set; }
    }
}
